### DISCUSSION TITLE
How to communicate betweens Screens

### DISCUSSION THREAD

**Discussion by ghost** [Textualize/textual#2558] [Q&A] [answered]
How do you communicate between Screens?

I have a main Screen pushed by the app that contains a TextLog.
I want to write to this textlog from other Screens.

I tried the following:

```
from textual.message import Message


class LogMessage(Message):
    """Color selected message."""

    def __init__(self, message) -> None:
        self.message = message
        super().__init__()
```
On the screen that sends the log message:
```
self.post_message(LogMessage("Logged Out"))
self.app.pop_screen()
```
On the main screen:
```   
    async def on_log_message(self, message: LogMessage):
        self.query_one("#log", TextLog).write(message)
```

The TextLog has id "log".

But nothing appears on the Log

The only workaround I have found is to use self.app.post_message and put the on_log_message method into the App.
Is this how it is supposed to be done?

**Comment by davep** [comment, marked as answer]
You may find that the answer I gave in https://github.com/Textualize/textual/discussions/2559 will be helpful with this too.

To directly answer the question here though: the best thing to do is to post the message to the screen or widget that is interested in it; don't post it to the screen that's about the be popped. So something more like:

```python
self.app.screen.post_message(LogMessage("Logged Out"))
self.app.pop_screen()
```

As it is though, the callback system mentioned in the other similar question you asked is likely going to be a more satisfying approach.

**Comment by ghost** [comment]
Thanks for the answer, was able to work things out thanks to it!
